pkg/controller/rmdworkload/rmdworkload_controller.go

```diff
@@ -143,40 +137,23 @@ func (r *ReconcileRmdWorkload) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	// Create a map of node state namespaces for RMD pod discovery.
-	// Nodestates and RMD pods are created with the same namespace by node_controller.
-	// This is the same namespace as the parent node, or default if not set.
-	rmdWorkloadName := rmdWorkload.GetObjectMeta().GetName()
-	nodeNamespaces := make(map[string]string)
-	for _, rmdNodeState := range rmdNodeStates.Items {
-		nodeNamespaces[rmdNodeState.Spec.Node] = rmdNodeState.GetObjectMeta().GetNamespace()
+	// Discover all RMD instances that the reconciled RmdWorkload is targeting.
+	// Add or Update those instances with the reconciled RmdWorkload accordingly
+	targetNodes, err := r.findTargetedNodes(request, rmdNodeStates, rmdWorkload)
```

Comment:
Nit: would change naming here from `targetNodes` to `targetedNodes` just to be consistent.